app/controllers/users_controller.rb

```diff
@@ -1,6 +1,14 @@
 class UsersController < ApplicationController
   def show
     @user = User.find(params[:id])
+    if not @user.is_instructor?
+      # List of this user's questions and answers
+      @answers = @user.answers.where.not(question_id: nil)
+      @questions = @answers.map{ |answer| answer.question }
```

Comment:
Honestly, not really. I don't have much experience dealing with joins. My solution in my PRs has also been hacky, but they work. We can get together to hash out a better solution time permitting.